.github/workflows/yaml_blueprint_to_java_template.yml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This workflow generates or modifies Java template files based on changes
+# to YAML files in a pull request.
+
+name: YAML to Java Template Generation or Modification
+
+on:
+  pull_request:
+    paths:
+      - 'yaml/src/main/yaml/*.yaml'
+
+permissions: write-all
+
+jobs:
+  job-generate-templates:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5.0.0
+        with:
+          fetch-depth: 0
```

Comment:
Any reason for this? This will slow clones

Reply:
When I was iterating on this and with my understanding, it was required to have the full history to know about the branch and its context for pushing back to the same PR.  Its a mute point now based on other comments, so will remove. Thanks.